### PR TITLE
Gray validate dialogue if no modification made

### PR DESCRIPTION
**Pull request by anas-drissi**

### CODE REVIEW

File: src/components/dialogs/inputs/input-hooks.js [hunk 1]
```diff
@@ -331,17 +335,21 @@ export const useEnumValue = ({
     getEnumLabel = getLabel,
 }) => {
     const [value, setValue] = useState(defaultValue);
-
+    const [hasChanged, setHasChanged] = useState(false);
```
Comment (etiennehomer) [inline src/components/dialogs/inputs/input-hooks.js:338]: Not used. You can remove it

File: src/components/dialogs/inputs/input-hooks.js [hunk 2]
```diff
@@ -399,7 +407,7 @@ export const useEnumValue = ({
         setValue(defaultValue || '');
     }, [defaultValue]);
 
-    return [value, field];
+    return [value, field, hasChanged, setHasChanged];
```
Comment (etiennehomer) [inline src/components/dialogs/inputs/input-hooks.js:410]: Can be removed

File: src/components/dialogs/inputs/use-autocomplete-field.js [hunk 3]
```diff
@@ -95,7 +95,7 @@ export const useAutocompleteField = ({
     const [userStr, setUserStr] = useState('');
     const [value, setValue] = useState(defaultValue);
     const [error, setError] = useState();
-
+    const [hasChanged, setHasChanged] = useState(false);
```
Comment (etiennehomer) [inline src/components/dialogs/inputs/use-autocomplete-field.js:98]: Same. Not used

File: src/components/dialogs/inputs/use-autocomplete-field.js [hunk 4]
```diff
@@ -322,5 +326,5 @@ export const useAutocompleteField = ({
         userStr,
     ]);
 
-    return [value, field, setValue];
+    return [value, field, setValue, hasChanged, setHasChanged];
```
Comment (etiennehomer) [inline src/components/dialogs/inputs/use-autocomplete-field.js:329]: Same. Not used

**Review by etiennehomer — CHANGES_REQUESTED** [review]

File: src/components/dialogs/inputs/use-autocomplete-field.js (hunk 4 above)
Comment (anas-drissi) [reply src/components/dialogs/inputs/use-autocomplete-field.js:329]: It's fixed.

File: src/components/dialogs/inputs/use-autocomplete-field.js (hunk 3 above)
Comment (anas-drissi) [reply src/components/dialogs/inputs/use-autocomplete-field.js:98]: It's fixed.

File: src/components/dialogs/inputs/input-hooks.js (hunk 2 above)
Comment (anas-drissi) [reply src/components/dialogs/inputs/input-hooks.js:410]: It's fixed.

**Comment by etiennehomer** [comment]
Tests ok, code ok

**Review by bendaamerahm — APPROVED** [review]
test: ok

**Pull request merged**